third_party/cyw30739_sdk/compress_cgs_data.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python
+
+import argparse
+import io
+import pathlib
+import re
+import subprocess
+import sys
+
+from collections import namedtuple
+
+
+Item = namedtuple('Item', 'line type nested_data')
+Entry = namedtuple('Entry', 'items name remark')
+CommentedBytes = namedtuple('CommentedBytes', 'items data')
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--output", required=True, type=pathlib.Path)
+    parser.add_argument("--input", required=True, type=pathlib.Path)
+    parser.add_argument("--lzss_tool", required=True, type=pathlib.Path)
+
+    option = parser.parse_args()
+
+    cgs = parse_cgs(option.input)
+    compress_cgs(option, cgs)
+    export_cgs(option.output, cgs)
+
+
+def parse_cgs(file_name):
+    cgs = []
+
+    with open(file_name, mode="r") as file:
+        while True:
```

Comment:
why not `for line in file.readlines()` ?

Reply:
fixed.